### PR TITLE
ZCS-12373: WCAG | Low complexity Aria issues

### PR DESCRIPTION
**Pull request by miteshsavani810**
Resolved following issues.

- **Issue 1:  1.1.1_CalendarImagedoesn'tHaveTextAlternative.docx**
   > Folder (i.e. Inbox, Calendar, Tasks, etc.) image doesn't have alternative text.
- **Issue 2: 1.3.1_aria-label and visual name is not matching.docx**
  > New dropdown menu (New Message, New Contact, New Appointment, etc.) contains aria-label which is not matching with visual name.
- **Issue 3: 1.3.1_NextandPrevious Week buttons are not announcing the updated value.docx**
  > On Calendar view, On navigation to Next/Prev button, SR not announcing updated value.
- **Issue 4: 1.3.1_aria issue-Mail.docx**
  > Mail list having issue of `aria-multiselectable` and `role`. These attributes are not assign to correct element. (Notes. The same issue is also present of Tasks and Briefcase list).
- **Issue 5: 1.3.1_aria auto issue-NewMail.docx**
  > Auto complete elements (To, Cc, Bcc fields) has contains `aria-autocomplete` attribute with wrong value.
- **Issue 6: 1.3.1_aria issue-ContactGroup.docx**
  > On Contact group creation screen, Both Members list and contact list having issue of `aria-multiselectable` and `role`. These attributes are not assign to correct element.

See the changes of zm-ajax on this pull request #https://github.com/Zimbra/zm-ajax/pull/100

### CODE REVIEW

File: WebRoot/js/zimbraMail/abook/view/ZmGroupView.js [hunk 1]
```diff
@@ -1556,6 +1556,7 @@ ZmGroupListView = function(parent) {
 	DwtListView.call(this, {parent:parent, className:"DwtChooserListView ZmEditGroupContact",
 							headerList:this._getHeaderList(parent), view:this._view, posStyle: Dwt.RELATIVE_STYLE});
 	Dwt.setScrollStyle(this._elRef, Dwt.CLIP);
+	this._listDiv.setAttribute("aria-label", ZmMsg.memberList);
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/abook/view/ZmGroupView.js:1559]: isn't DwtListView setting this value?

File: WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js [hunk 2]
```diff
@@ -48,6 +48,7 @@ ZmDetailListView = 	function(parent, controller, dropTgt) {
 				  controller:controller, headerList:headerList, dropTgt:dropTgt};
 	ZmBriefcaseBaseView.call(this, params);
 
+	this._listDiv.setAttribute("aria-label", ZmMsg.documentList);
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js:51]: same as above if DwtListView can set this based on some passed param then we don't have to repeat this statement in all list views

File: WebRoot/js/zimbraMail/mail/view/ZmMailMsgListView.js [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
- *
- * Software distributed under the License is distributed on an "AS IS" basis,
- * WITHOUT WARRANTY OF ANY KIND, either express or implied.
- * See the License for the specific language governing rights and limitations under the License.
- * The Original Code is Zimbra Open Source Web Client.
- * The Initial Developer of the Original Code is Zimbra, Inc.  All rights to the Original Code were
- * transferred by Zimbra, Inc. to Synacor, Inc. on September 14, 2015.
- *
- * All portions of the code are Copyright (C) 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016 Synacor, Inc. All Rights Reserved.
+ * Zimbra Collaboration Suite Web Client
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/mail/view/ZmMailMsgListView.js:3]: not sure what changes in license block

File: WebRoot/js/zimbraMail/share/model/ZmOrganizer.js [hunk 4]
```diff
@@ -1038,6 +1038,16 @@ function() {
...
 };
 
+/**
+ * Gets the icon alter text.
+ * 
+ * @return	{String}	the icon alter text
+ */
+
+ZmOrganizer.prototype.getIconAlterText = function() {
+	return this.name + " " + ZmMsg.icon;
```
Comment (HimanshuSojitraSynacor) [inline WebRoot/js/zimbraMail/share/model/ZmOrganizer.js:1048]: The word "icon" will be redundant if SR will read out "An image of calendar icon". In that case let's just have "Calendar" as alt text?

File: WebRoot/js/zimbraMail/mail/view/ZmMailMsgListView.js (hunk 3 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/mail/view/ZmMailMsgListView.js:3]: There is no any text change. I tried by replacing with original text but it doesn't get resolved.

File: WebRoot/js/zimbraMail/share/model/ZmOrganizer.js (hunk 4 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/share/model/ZmOrganizer.js:1048]: I checked with NVDA on windows machine. SR is reading out as "Calendar icon", "Inbox icon".
image

File: WebRoot/js/zimbraMail/share/controller/ZmTreeController.js [hunk 5]
```diff
@@ -546,7 +546,7 @@ function(treeItem) {
  */
 ZmTreeController.prototype._setTreeItemColor =
 function(treeItem, organizer) {
-	treeItem.setImage(organizer.getIconWithColor());
+	treeItem.setImage(organizer.getIconWithColor(), organizer.getIconAlterText());
```
[the remark this answers is not in the document]
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/share/controller/ZmTreeController.js]: Fixed

File: WebRoot/js/zimbraMail/abook/view/ZmGroupView.js (hunk 1 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/abook/view/ZmGroupView.js]: Done.

File: WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js (hunk 2 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/view/ZmDetailListView.js]: Done.

**Review by silentsakky — APPROVED** [review]

**Pull request merged**